1512 modify claim page texts

StanislavShevchenko opened the pull request:
# Description

**Parent branch 1450_1432_User_should_see_new_claim_screen**

settings for phase0 and phase1 need set to env REACT_APP_ENV_PHASE_ZERO 
REACT_APP_ENV_PHASE_ZERO = true - phase0
REACT_APP_ENV_PHASE_ZERO = false - phase1

Modify Claim page texts

About #1512 

# How Has This Been Tested?

Go to Claim screen

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [x] I have added screenshots related to my pull request ( for frontend tasks)
- [x] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

**Phase - 0**
![phase-0](https://user-images.githubusercontent.com/8187382/78345948-04476f00-75a7-11ea-8445-770862c5af7e.gif)

**Phase - 1**
![phase-1](https://user-images.githubusercontent.com/8187382/78346000-19240280-75a7-11ea-8fb6-ce51c1d1dd87.gif)

serdiukov-o-nordwhale commented:
Please, also provide how to configure switching between phases (in the task definition)

serdiukov-o-nordwhale commented on src/components/dashboard/Claim/Phase0.js:
```diff
@@ -0,0 +1,119 @@
+          <ClaimButton
+            isCitizen={isCitizen}
+            entitlement={entitlement}
+            nextClaim={state.nextClaim}
```
pass just pros used.
e.g. here pass just nextClaim, not whole 'state' prop

serdiukov-o-nordwhale replied:
what is the state and why it's needed ? why not just pass entitlement, nextClaim isCitizen etc ?

serdiukov-o-nordwhale commented on src/components/dashboard/ClaimButton.js:
```diff
@@ -1,41 +1,52 @@
-  </>
-)
+const ButtonAmountToClaim = ({ entitlement, isCitizen, styles }) =>
+  Config.claimContentPhaseZero ? (
```
do not use config here. your kind of design should be defined **on claim screen only**
add some prop e.g. displayNextClaim: boolean and pass it dependently of Config.claimContentPhaseZero **from Claimphase1 component**

StanislavShevchenko commented:
> # Description
> **Parent branch 1450_1432_User_should_see_new_claim_screen**
> 
> Modify Claim page texts
> 
> About #1512
> 
> # How Has This Been Tested?
> Go to Claim screen
> 
> # Checklist:
> * [x]  PR title matches follow: (Feature|Bug|Chore) Task Name
> * [x]  My code follows the style guidelines of this project
> * [x]  I have followed all the instructions described in the initial task (check Definitions of Done)
> * [x]  I have performed a self-review of my own code
> * [x]  My changes generate no new warnings
> * [x]  I have added tests that prove my fix is effective or that my feature works
> * [x]  New and existing unit tests pass locally with my changes
> * [x]  I have added reference to a related issue in the repository
> * [x]  I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
> * [x]  I have added screenshots related to my pull request ( for frontend tasks)
> * [x]  I have pasted a gif showing the feature.
> * [x]  @mentions of the person or team responsible for reviewing proposed changes
> 
> **Phase - 0**
> ![phase-0](https://user-images.githubusercontent.com/8187382/78345948-04476f00-75a7-11ea-8445-770862c5af7e.gif)
> 
> **Phase - 1**
> ![phase-1](https://user-images.githubusercontent.com/8187382/78346000-19240280-75a7-11ea-8fb6-ce51c1d1dd87.gif)

StanislavShevchenko commented:
@serdiukov-o-nordwhale 
> Please, also provide how to configure switching between phases (in the task definition)

done, added to description

serdiukov-o-nordwhale commented on src/components/dashboard/Claim/Phase1.js:
```diff
@@ -0,0 +1,65 @@
+            <Image source={arrowsDown} style={styles.arrowsDown} />
+          </Section.Row>
+        </Section.Stack>
+        <Section.Stack style={styles.btnBlock}>
```
as btn block is common and contains some logic let's move  it to component:

```
const ButtonBlock = ({ styles, entitlement, isCitizen, nextClaim, handleClaim, faceRecognition, ...props }) => {
  const onPress = useCallback(() => {
    if (!isCitizen) {
      faceRecognition();
      return
    }

    if (entitlement) {
      handleClaim()
    }
  }, [entitlement, isCitizen, faceRecognition, handleClaim])

  return (
    <Section.Stack style={styles.btnBlock}>
           <ClaimButton
             isCitizen={isCitizen}
             entitlement={entitlement}
             nextClaim={nextClaim}
             onPress={onPress}
             {...props}
           />
         </Section.Stack>
  )
}
```

serdiukov-o-nordwhale commented on src/components/dashboard/Claim/Phase0.js:
```diff
@@ -0,0 +1,119 @@
+                  fontSize={18}
+                  fontWeight="bold"
+                  fontFamily="slab"
+                  onPress={() => window.open(Config.urlToLearnMore, '_blank')}
```
Won't work on native, please, use https://reactnative.dev/docs/linking abstraction

serdiukov-o-nordwhale replied:
and wrap to callback pls or move outside component as it's props-independent

LiavGut commented:
@StanislavShevchenko Couple of thinks:
1. The white line at the bottom of the background is not in the correct width.
2. The background should be lowered by 2 pixels so the color switch will look seamless.
3. Is the rectangle with the daily G$ responsive and will change according to the amount? if not it should be as per design.
4. the font text is not the correct one.

If it's a work in progress then forget about it :)

StanislavShevchenko commented:
@LiavGut 
> 
> 1. The white line at the bottom of the background is not in the correct width.

- This is the line of background from the SVG, we do not control it. In reality, it looks more
![image](https://user-images.githubusercontent.com/8187382/78388324-7640a800-75e9-11ea-8153-9ad613c64835.png)

> 2. The background should be lowered by 2 pixels so the color switch will look seamless.
- done 

> 3. Is the rectangle with the daily G$ responsive and will change according to the amount? if not it should be as per design.

- responsive and will change according to the amount

> 4. the font text is not the correct one.

- done 

Thanks!

StanislavShevchenko commented:
### **Phase 0** 
![phase0-2](https://user-images.githubusercontent.com/8187382/78391068-552e8600-75ee-11ea-9297-a708a01a9864.gif)
### **Phase 1** 
![phase1-2](https://user-images.githubusercontent.com/8187382/78391096-62e40b80-75ee-11ea-963e-78d0ec7671d4.gif)
